src/dbl_lnk_lst.py

```
@@ -83,43 +88,54 @@ def size(self):
         current = self.head
-        while current.data is not None:
+        while current is not None:
             if current.data == val:
                 return current
-            elif current.next_node is None:
-                return None
             current = current.next_node
+        return None
 
     def remove(self, val):  # if last node --> set Node to None
         """Remove the given node from the list, wherever it might be."""
         node = self.search(val)
 
-        if node.next_node is not None:
+        try:
             node.data = node.next_node.data
             node.next_node = node.next_node.next_node
-        elif node.next_node is None:
+        except AttributeError:
             self.tail = node.prev_node
             node.data = None
-            node.prev_node.next_node = None
+            try:
+                node.prev_node.next_node = None
+            except AttributeError:
+                if self.length == 0:
+                    return self
 
         self.length -= 1
-        return (node, self, self.length)
+        return self
 
     def append(self, val):
         """Append the val arg to the end of the list as a new node."""
         self.tail.next_node = Node_Dll(val, self.tail, None)
         self.tail = self.tail.next_node
+        self.length += 1
+        return self
 
     def shift(self):
         """
         Shift the node off the tail of the list and return the value of the
         node.
         """
-        if self.length > 0:
+        if self.length > 1:
             shifted_node = self.tail
             self.tail = self.tail.prev_node
             self.tail.next_node = None
             self.length -= 1
             return shifted_node.data
+        elif self.length == 1:
+            shifted_node_data = self.tail.data
+            self.head.data = None
+            self.tail.data = None
+            self.length = 0
+            return shifted_node_data
         else:
```

`pop` and `shift` should raise the appropriate exception when popping or shifting from an empty list. I know this isn't made explicit in the assignment (it should be).